### PR TITLE
[New Rule] Pspy Process Monitoring Detected

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This rule leverages auditd to monitor for processes scanning different processes within the /proc directory using the openat syscall. This is a strong indication for the usage of the pspy utility. Attackers may leverage the pspy process monitoring utility to monitor system processes without requiring root permissions, in order to find potential privilege escalation vectors.

The non-ecs-schema.json file had to be editted for this rule to pass unit testing. This file was also altered for the [UDP reverse shell rule](https://github.com/elastic/detection-rules/pull/2906/files), but because this one will be merged earlier, I will remove the changes to this file from that PR. 

## Detection
The rule was tested with both 64 and 32 bit (different a0 and a2 args for different archs), and on different unix distributions to ensure that the arguments passed to the openat syscall were the same (which makes sense, but tested anyways). 

```
sequence by process.pid, host.id with maxspan=5s
[ file where host.os.type == "linux" and event.dataset == "auditd_manager.auditd" and 
  auditd.data.syscall == "openat" and file.path == "/proc" and auditd.data.a0 : ("ffffffffffffff9c", "ffffff9c") and 
  auditd.data.a2 : ("80000", "88000") ] with runs=10
```
image

### CODE REVIEW

File: rules/linux/discovery_pspy_process_monitoring_detected.toml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+[rule]
+author = ["Elastic"]
+description = """
+This rule leverages auditd to monitor for processes scanning different processes within the /proc directory using the 
+openat syscall. This is a strong indication for the usage of the pspy utility. Attackers may leverage the pspy process 
+monitoring utility to monitor system processes without requiring root permissions, in order to find potential privilege 
+escalation vectors.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-auditd_manager.auditd-*"]
```
Comment (w0rk3r) [inline rules/linux/discovery_pspy_process_monitoring_detected.toml:18]:
```suggestion
index = ["logs-auditd_manager.auditd-*"]
```
As it filters on `event.dataset == "auditd_manager.auditd"`

File: rules/linux/discovery_pspy_process_monitoring_detected.toml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+-w /proc/ -p r -k audit_proc
+```
+
+Add the newly installed `auditd manager` to an agent policy, and deploy the agent on a Linux system from which auditd log files are desirable.
+"""
+references = ["https://github.com/DominicBreuker/pspy"]
+risk_score = 21
+rule_id = "bdb04043-f0e3-4efa-bdee-7d9d13fa9edc"
+severity = "low"
+tags = ["OS: Linux", "Use Case: Threat Detection", "Tactic: Discovery"]
```
Comment (w0rk3r) [inline rules/linux/discovery_pspy_process_monitoring_detected.toml:46]:
```suggestion
tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Discovery"]
```

File: rules/linux/discovery_pspy_process_monitoring_detected.toml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+[[rule.threat.technique]]
+id = "T1082"
+name = "System Information Discovery"
+reference = "https://attack.mitre.org/techniques/T1082/"
+
+[rule.threat.tactic]
+id = "TA0007"
+name = "Discovery"
+reference = "https://attack.mitre.org/tactics/TA0007/"
```
Comment (w0rk3r) [inline rules/linux/discovery_pspy_process_monitoring_detected.toml:71]:
```suggestion
reference = "https://attack.mitre.org/tactics/TA0007/"

```

**Review by w0rk3r — APPROVED** [review]
🧙🏼

File: rules/linux/discovery_pspy_process_monitoring_detected.toml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+name = "Potential Pspy Process Monitoring Detected"
+note = """## Setup
+This rule requires the use of the `auditd_manager` integration. `Auditd_manager` is a tool designed to simplify and enhance the management of the audit subsystem in Linux systems. It provides a user-friendly interface and automation capabilities for configuring and monitoring system auditing through the auditd daemon. With `auditd_manager`, administrators can easily define audit rules, track system events, and generate comprehensive audit reports, improving overall security and compliance in the system. The following steps should be executed in order to install and deploy `auditd_manager` on a Linux system. 
+
+```
+Kibana -->
+Management -->
+Integrations -->
+Auditd Manager -->
+Add Auditd Manager
```
Comment (terrancedejesus) [inline rules/linux/discovery_pspy_process_monitoring_detected.toml:30]: When the rule is built, `related_integrations` is added to the rule where `auditd_manager` and a compatible version is added. This is then used by the Kibana UI to show what integration needs installed and has a hyperlink to the integrations page for auditd manager. 

Doesn't hurt to include it in setup as well, but wanted to share this.

**Review by terrancedejesus — APPROVED** [review]
LGTM, great work!

File: rules/linux/discovery_pspy_process_monitoring_detected.toml (hunk 4 above)
Comment (Aegrah) [reply rules/linux/discovery_pspy_process_monitoring_detected.toml:30]: Thanks for sharing, will do an overhaul of all of the setup guides I wrote for auditd_manager in a seperate PR when the docs team has looked at it as well. Will add this to my notes.

**Pull request merged**